sources/modules/compilation-model/headers.md

```diff
@@ -0,0 +1,105 @@
+## Module name: topic name
+
+_Skeleton descriptions are typeset in italic text,_
+_so please don't remove these descriptions when editing the topic._
+
+### Overview
+
+_Provides a short natural language abstract of the module’s contents._
+_Specifies the different levels of teaching._
+
+------------------------------------------------------------------------
+Level             Objective
+----------------- ------------------------------------------------------
+Foundational      Including code through standard library headers
+
+Main              Organize function and class declarations into reusable
+                  files for inclusion into others
+
+Advanced          --- (inlining? templates? hygiene, such as using namespace?)
+
+------------------------------------------------------------------------
+
+### Motivation
+
+_Why is this important?_
+_Why do we want to learn/teach this topic?_
+
+Understanding and using header files allows for common declarations to 
+be used across multiple files without copying-and-pasting the text. Also,
+understanding headers allows re-use of most existing 3rd-party libraries,
+as well as the C++ standard library.
+
+### Topic introduction
+
+_Very brief introduction to the topic._
+
+The contents of header files are injected into a source file to allow 
+multiple distinct source files to share a single source of common 
+declarations. 
+
+### Foundational: Including code through standard library headers
+
+#### Background/Required Knowledge
+
+A student is able to:
+
+* Invoke the compiler on a single source file to build an executable
+
+
+#### Student outcomes
+
+_A list of things "a student should be able to" after the curriculum._
+_The next word should be an action word and testable in an exam._
+_Max 5 items._
+
+A student should be able to:
+
+1. Include common standard library headers to get access to commonly used declarations
+
+#### Caveats
+
+_This section mentions subtle points to understand, like anything resulting in
+implementation-defined, unspecified, or undefined behavior._
+
+#### Points to cover
+
+_This section lists important details for each point._
+
+* Usage of standard library code requires inclusion of various header files
+* Give examples of various header files and when they may be needed for inclusion
+* Include links to reference materials for finding out what headers are used for various feature sets
+
+### Main: Organizing function and class declarations for reuse
+
+#### Background/Required Knowledge
+
+* All of the above.
```

Comment:
Is "All of the above" on line 77 referring to all of the foundational background?  I think that it would be clearer to say this explicitly.  Incidentally, is it implied that all background for a topic at the foundational level is also required for the main level?  If so, then perhaps such a statement is unnecessary?